Radiation pulse changes. The chance of getting irradiated decreases the further you are from the source, and from objects that block radiation.

Pickle-Coding opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Radiation pulse functionality has changed. Radiation pulses will now use intensity to determine chance of getting irradiated. Intensity decreases the further the target is from the radiation source, and decreases from radiation insulating objects blocking the path between the target and the source. The intensity increases the chance of getting irradiated exponentially.

Most radiation pulses have been changed a bit for this functionality. If you are in range of the pulse, but far away, you are less likely to get irradiated, while being right next to it will have a high change of irradiation. Being behind walls will also lower the chance of getting irradiated, even if it doesn't completely block the radiation.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Currently the effects of radiation pulses intensity are completely uniform, except in some cases where walls can block them. This makes the areas of what is considered a safe area and not safe area very binary, and doesn't look natural. Giving it diminishing returns over range is more intuitive, as the transition between a radioactive area and a not radioactive area is smoother. It allows for large radiation pulses (I am not increasing their range, just talking about the existing ones like the supermatter) to be very dangerous when you are close to the source, but not so extreme if you just crossed the border. Large radiation pulses with not low radiation chance made it frustrating when you crossed the boundaries and you get irradiated quickly.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: The further you are from a radiation pulse, and the more objects that block it, the lower your odds are to get irradiated.
balance: Most radiating emitting things have their radiation pulses functionality changed. Being near them is more dangerous, and being far away is less dangerous.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pickle-Coding committed:
Radiation.

Replaces radiation chance with radiation intensity on radiation pulses. Radiation intensity determines the chance of getting irradiated, and diminishes over range or getting blocked by objects.

Getting close to the radioactive source will give you a high chance to get irradiated, while being far but still in range will give you a low chance to get irradiated.

Pickle-Coding commented:
Drafting until I test all the radiating emitting objects to see if their intensity are at the correct values.

Pickle-Coding committed:
I suck at spelling.

Balances some atmos related radiation stuff. Changes supermatter radiation stuff, though I would probably need to adjust the threshold to adjust for the new changes.

Calling radiation pulse without intensity parameter filled will now set the intensity to be the value where it will give an object a 5% chance to get irradiated from the max range of the pulse, assuming there are no objects between it that absorb radiation. The same will happen if you call radiation pulse without max range parameter filled.

Mothblocks commented on code/controllers/subsystem/radiation.dm:
```diff
@@ -57,6 +59,18 @@ SUBSYSTEM_DEF(radiation)
 		if (current_insulation <= pulse_information.threshold)
 			continue
 
+		if(isnull(pulse_information.max_range) && pulse_information.intensity) // If max range is undefined, then set it to the distance where the intensity drops to the constant.
```
I'm not a fan of this. Intensity is a metric with no obvious meaning, I intentionally removed it with the new system.

It's not clear to feature coders or maintainers what an "intensity" number actually means, at least not without looking everywhere else for context. It is obvious what max range is.

Pickle-Coding replied:
Could remove that if you don't want intensity to be used like that.

Mothblocks reviewed (changes requested):
I'm not convinced on why this is a good idea--it looks like you are taking a concept that is very easy to review: max range and chance, and making it into a magic number that is now impossible for me to reason about.

Pickle-Coding committed:
Adds more documentation.

Pickle-Coding commented on code/__HELPERS/radiation.dm:
```diff
@@ -17,9 +17,9 @@
 /proc/radiation_pulse(
 	atom/source,
-	max_range,
+	max_range = null,
```
Allows people to call radiation pulse without max_range defined in the parameters. Maybe I could not allow that since you aren't really a fan of people just using the intensity.

Mothblocks replied:
`max_range` and `max_range = null` mean the exact same thing.

Pickle-Coding commented on code/__HELPERS/radiation.dm:
```diff
@@ -17,9 +17,9 @@
+	max_range = null,
 	threshold,
-	chance = DEFAULT_RADIATION_CHANCE,
+	intensity = null,
```
I set this to null so people can call a radiation pulse without defining intensity, hopefully to make it easier to balance if radiation pulses are more uniform and people don't know what to set the intensity to.

Pickle-Coding commented:
I'll try to make intensity easily understandable with documentation. Alternatively, intensity could not be a parameter, but is defined from the chance parameter so it can do the math properly to diminish returns.

Maybe have a chance parameter instead of intensity that defines the chance to get irradiated half way between the epicentre and the edge of the pulse?

Pickle-Coding committed:
Undefined.

Pickle-Coding committed:
You have to use max range parameter.

Mothblocks commented:
The important bit here is that consumers of `radiation_pulse` are extremely easy for a maintainer to read.

If the internal fiddly bits use some different system, that's okay.

Pickle-Coding committed:
It is better this way.

Intensity is no longer a parameter. Brings back using chance as a parameter, and it determines the chance of getting irradiated if the thing is at a distance of half the max range of a radiation pulse.

Tritium fire radiation pulses have been changed. More consistent with other gas reaction radiation pulses.

Pickle-Coding commented:
I removed intensity parameter and brought back chance parameter. The chance parameter will determine the chance to get irradiated if the target's distance is half of the max tange.

Pickle-Coding commented:
Might try to improve supermatter radiation so at normal power it will have a low chance to irradiate people in engine room, and high chance if in chamber, then I might open the pull request.

Pickle-Coding committed:
Fix comment.

Pickle-Coding commented:
I was dumb.

Pickle-Coding commented:
Upon testing I think supermatter radiation is okay and doesn't need change (though it could). I'll open this now.

Mothblocks commented:
Seeing as consumers are unaffected, this is okay now
